src/System.Management.Automation/FormatAndOutput/common/ListWriter.cs

```diff
@@ -220,14 +228,30 @@ private void WriteSingleLineHelper(string prependString, string line, LineOutput
             // display the string collection
             for (int k = 0; k < sc.Count; k++)
             {
+                string str = sc[k];
+                _cachedBuilder.Clear();
+
                 if (k == 0)
                 {
-                    lo.WriteLine(PSStyle.Instance.Formatting.FormatAccent + prependString + PSStyle.Instance.Reset + sc[k]);
+                    _cachedBuilder
+                        .Append(PSStyle.Instance.Formatting.FormatAccent)
+                        .Append(prependString)
+                        .Append(PSStyle.Instance.Reset)
+                        .Append(str);
                 }
                 else
                 {
-                    lo.WriteLine(padding + PSStyle.Instance.Formatting.FormatAccent + PSStyle.Instance.Reset + sc[k]);
+                    _cachedBuilder
+                        .Append(padding)
+                        .Append(str);
                 }
+
+                if (str.Contains(ValueStringDecorated.ESC) && !str.EndsWith(PSStyle.Instance.Reset))
```

Comment:
What if the ESC value is escaped?

Reply:
> What if the ESC value is escaped?

Can you escape a literal escape? Genuine question

Reply:
@iSazonov can you show an example of that? Currently, this check is at the core of VT handling in PowerShell (`$PSStyle` support depends on it).

Reply:
Backtick doesn't work at the level?

Reply:
Nah it only works at the parser level to escape `` `e``. Once it's a literal `0x1b` char it doesn't do anything